lib/AppInfo/Application.php

```diff
@@ -70,9 +71,19 @@ private function registerMasterHooks(IAppContainer $c) {
 		$master = $c->query(Master::class);
 		Util::connectHook('OC_User', 'pre_login', $master, 'handleLoginRequest');
 
-		/** @var IEventDispatcher $eventDispatcher */
-		$eventDispatcher = $c->getServer()->query(IEventDispatcher::class);
-		$eventDispatcher->addListener(AddContentSecurityPolicyEvent::class, AddContentSecurityPolicyListener::class);
+		try {
+			/** @var IEventDispatcher $eventDispatcher */
+			$eventDispatcher = $c->getServer()->query(IEventDispatcher::class);
+			$eventDispatcher->addListener(
+				AddContentSecurityPolicyEvent::class,
+				function (AddContentSecurityPolicyEvent $event) {
+					/** @var AddContentSecurityPolicyListener $listener */
+					$listener = \OC::$server->get(AddContentSecurityPolicyListener::class);
+					$listener->handle($event);
+				}
+			);
```

Comment:
Right, I think I mixed that up with the method on IRegistrationContext

Maybe it should just use addServiceListener then:

```suggestion
			$eventDispatcher->addServiceListener(AddContentSecurityPolicyEvent::class, AddContentSecurityPolicyListener::class);
```

Reply:
yes, should work too. Haven't tested, but we need to choose one at least :) 

Create your PR if you want to implement your solution, i'll test & vouch

Reply:
Sure, pushed to https://github.com/nextcloud/globalsiteselector/pull/52 :) testing would be very welcome